builtin/providers/aws/resource_aws_dynamodb_table_migrate.go

```diff
@@ -0,0 +1,69 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/terraform"
+	"strings"
+)
+
+func resourceAwsDynamoDbTableMigrateState(
+	v int, is *terraform.InstanceState, meta interface{}) (*terraform.InstanceState, error) {
+	switch v {
+	case 0:
+		log.Println("[INFO] Found AWS DynamoDB Table State v0; migrating to v1")
+		return migrateDynamoDBStateV0toV1(is)
+	default:
+		return is, fmt.Errorf("Unexpected schema version: %d", v)
+	}
+}
+
+func migrateDynamoDBStateV0toV1(is *terraform.InstanceState) (*terraform.InstanceState, error) {
```

Comment:
Hey @catsby 
As discussed together, made a first shot of the state migration for a set.

In order to check the overall work, I provisioned a DDB table using TF 0.8.8, then compiled this version of TF.
1. A plan didn't throw any diff
2. When updating read/write capacities for GSIs, diffs were all ok

Starting at line 35, I do think there is a better way to extract set hashes and add new entries... but this may require more complex work, using the Eval*.go work perhaps?

Still, a big thank for the inputs you gave me, was really helpful!